git/keep-feature-branches-up-to-date.md

```diff
@@ -0,0 +1,25 @@
+# How To: Keep Feature Branches Up To Date
+
+If commits come into `upstream/master` after your branch off, you will need to keep your branch up to date with `upstream/master` before you PR. To do this, I recommend the slightly more challenging, yet more simple `rebase` method.
+
+## Step 1: Ensure you're on the correct branch
+
+```bash
+git status
+# check current branch, is it correct? If not:
+git checkout my-feature-branch
+```
+
+## Step 2: Rebase!
+
+Rebase is a utility that takes the master branch and "plays your work over it". It is the cleanest way to update your code with the latest master.
+
+```bash
+git rebase -i upstream/master
+# you will be brought to an editor, exit and save
+# if you stop at a commit, it's probably a merge conflict. You can try to resolve with:
+# git mergetool
+# this will be a somewhat rare occurrence
+# After it says success, VERIFY everything looks good, then force push like so:
+git push -f # note: THIS DESTROYS OLD DATA ON GITHUB AND PUSHES YOUR LOCAL BRANCH UP FORCEFULLY
```

Comment:
This seems like a dangerous habit to get folks started on when they're still learning git. Why do they need to `push -f` at this point in the process? Have they been pushing commits as they go? If so, it might be worth explaining the ramifications of `push -f`, rather than just what it does: i.e. any other copies of this branch on other people's machines will also have to be updated.